internal/backend/remote-state/azure/auth/cli_auth.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package auth
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/opentofu/opentofu/internal/tfdiags"
+)
+
+type AzureCLIAuthConfig struct {
+	CLIAuthEnabled bool
+}
+
+type azureCLICredentialAuth struct{}
+
+func (cred *azureCLICredentialAuth) Name() string {
+	return "Azure CLI Auth"
+}
+
+func (cred *azureCLICredentialAuth) Construct(_ context.Context, config *Config) (azcore.TokenCredential, error) {
+	// The SubscriptionID and TenantID can be empty, and the logic of this will still be okay
+	return azidentity.NewAzureCLICredential(&azidentity.AzureCLICredentialOptions{
+		Subscription: config.StorageAddresses.SubscriptionID,
+		TenantID:     config.StorageAddresses.TenantID,
+	})
+}
+func (cred *azureCLICredentialAuth) Validate(config *Config) tfdiags.Diagnostics {
+	var diags tfdiags.Diagnostics
+	if !config.CLIAuthEnabled {
+		diags = diags.Append(tfdiags.Sourceless(
+			tfdiags.Error,
+			"Azure Backend: Command Line credentials",
+			"You set use_cli to false, which prevents the CLI Auth from being used. Remove this unless you're testing internals or otherwise explicitly want to prevent CLI Authentication.",
+		))
+		return diags
+	}
+	return diags
+}
+
+func (cred *azureCLICredentialAuth) AugmentConfig(config *Config) (err error) {
+	if config.StorageAddresses.SubscriptionID == "" {
+		config.StorageAddresses.SubscriptionID, err = getCliAzureSubscriptionID()
+		if err != nil {
+			return err
+		}
+	}
+	return checkNamesForAccessKeyCredentials(config.StorageAddresses)
+}
+
+type Subscription struct {
+	Id        string `json:"id"`
+	Name      string `json:"name"`
+	IsDefault bool   `json:"isDefault"`
+}
+
+// getCliAzureSubscriptionID obtains the subscription ID currently active in the
+// Azure profile. This assumes the user has the Azure CLI installed on their machine.
+func getCliAzureSubscriptionID() (string, error) {
+	rawSubscription, err := getCurrentSubscriptionInfo()
+	if err != nil {
+		return "", err
+	}
+
+	var subscription Subscription
+	err = json.Unmarshal(rawSubscription, &subscription)
+	if err != nil {
+		return "", fmt.Errorf("json error for azure subscription: %w", err)
+	}
+
+	return subscription.Id, nil
+}
+
+// getCurrentSubscriptionInfo is adapted from azure-sdk-for-go's CLI token retrieval
+func getCurrentSubscriptionInfo() ([]byte, error) {
+	ctx := context.Background()
+	commandLine := "az account show -o json"
+	var cliCmd *exec.Cmd
+	if runtime.GOOS == "windows" {
+		cliCmd = exec.CommandContext(ctx, "cmd.exe", "/c", commandLine)
```

Comment:
Is there a reason we are shelling out here instead of building os/args with exec.Command ?

Reply:
Not really. I was using the code in [azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go/blob/2ba09eb/sdk/azidentity/azure_cli_credential.go#L119-L172) (link goes to MIT-licensed code) as a basis for what the command should look like. I welcome any and all improvements to this!